src/librustc_mir/monomorphize/partitioning/mod.rs

```diff
@@ -0,0 +1,433 @@
+//! Partitioning Codegen Units for Incremental Compilation
```

Comment:
This comment block is taken verbatim from its original location. Parts of it describe the partitioning process over all and parts are specific to the algorithm (now called `DefaultPartitioner`). 

This could probably be left as is for now but eventually we should split it into two parts and move the bits about the algorithm into module documentation for the `DefaultPartitioner` algorithm.